enhancements/local-storage/automatic-detection.md

```diff
@@ -0,0 +1,349 @@
+but the disk/device paths needs to explicitly specified in the `LocalVolume` CR.
+The idea is to introduce a new feature in LSO
+  - Auto provisioning of localDevices
+    - This will involve introduction of a new CR called `LocalVolumeGroup`.
+    - The pupose of this will be to auto discover and provision PVs on devices which match the inclusion filter present in the CR.
+    - This will involve a continous process of discovery of devices via the diskmaker daemons. Any discovered devices which matches the inclusion filter will be considered for provisioning of PVs.
+
+## Design Details for `Auto provisioning of local devices`
+API scheme for `LocalVolumeGroups`:
+
+```go
+type LocalVolumeGroupList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []LocalVolumeGroup `json:"items"`
+}
+
+type LocalVolumeGroup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              LocalVolumeGroupSpec   `json:"spec"`
+	Status            LocalVolumeGroupStatus `json:"status,omitempty"`
+}
+
+type LocalVolumeGroupSpec struct {
+	// Nodes on which the automatic detection policies must run.
+	// +optional
+	NodeSelector *corev1.NodeSelector `json:"nodeSelector,omitempty"`
+	// StorageClassName to use for set of matched devices
+	StorageClassName string `json:"storageClassName"`
+	// MinDeviceCount is the minumum number of devices that needs to be detected per node.
+	// If the match devices are less than the minCount specified then no PVs will be created.
+	MinDeviceCount int `json:"minDeviceCount"`
+	// +optional
+	// Maximum number of Devices that needs to be detected per node.
+	MaxDeviceCount int `json:"maxDeviceCount"`
+	// VolumeMode determines whether the PV created is Block or Filesystem. By default it will
+	// be block
+	// + optional
+	VolumeMode PersistentVolumeMode `json:"volumeMode,omitempty"`
+	// FSType type to create when volumeMode is Filesystem
+	// +optional
+	FSType string `json:"fsType,omitempty"`
+	// If specified, a list of tolerations to pass to the discovery daemons.
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// DeviceInclusionSpec is the filtration rule for including a device in the device discovery
+	// +optional
+	DeviceInclusionSpec *DeviceInclusionSpec `json:"deviceInclusionSpec"`
+}
+
+// DeviceMechanicalProperty holds the device's mechanical spec. It can be rotational or nonRotational
+type DeviceMechanicalProperty string
+
+const (
+	// The mechanical properties of the devices
+	// Rotational refers to magnetic disks
+	Rotational DeviceMechanicalProperty = "Rotational"
+	// NonRotational refers to ssds
+	NonRotational DeviceMechanicalProperty = "NonRotational"
+)
+
+type DeviceType string
+
+const (
+	// The DeviceTypes that will be supported by the LSO.
+	// These Discovery policies will based on lsblk's type output
+	// Disk represents a device-type of disk
+	Disk DeviceType = "Disk"
```

Comment:
minor nit and we can address this during implementation review but IMO "disk" is bit vague when it comes to this. This would for example exclude a encrypted disk because iirc `lsblk` reports `crypt` as the TYPE.  It is fine to start with a small list and gradually expand but if we mean "disk" to be a whole disk then we should call out with a name that is more specific.

Reply:
Agreed. How about `RawDisk`?